pkg/k8s/config_map_cache_informer.go

```diff
@@ -0,0 +1,132 @@
+package k8s
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/informers"
+	infcorev1 "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+type InformerConfigMap struct {
+	lggr       logr.Logger
+	cmInformer infcorev1.ConfigMapInformer
+	bcaster    *watch.Broadcaster
+	cmObjCh    chan *corev1.ConfigMap
+}
+
+func (i *InformerConfigMap) MarshalJSON() ([]byte, error) {
+	lst := i.cmInformer.Lister()
+	cms, err := lst.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(&cms)
+}
+
+func (i *InformerConfigMap) Start(ctx context.Context) error {
+	i.cmInformer.Informer().Run(ctx.Done())
+	return errors.Wrap(
+		ctx.Err(), "configMap informer was stopped",
+	)
+}
+
+func (i *InformerConfigMap) Get(
+	ns,
+	name string,
+) (corev1.ConfigMap, error) {
+	cm, err := i.cmInformer.Lister().ConfigMaps(ns).Get(name)
+	if err != nil {
+		return corev1.ConfigMap{}, err
+	}
+	return *cm, nil
+}
+
+func (i *InformerConfigMap) Watch(
+	ns,
+	name string,
+) watch.Interface {
+	return watch.Filter(i.bcaster.Watch(), func(e watch.Event) (watch.Event, bool) {
+		cm := e.Object.(*corev1.ConfigMap)
+		if cm.Namespace == ns && cm.Name == name {
+			return e, true
+		}
+		return e, false
+	})
+}
+
+func (i *InformerConfigMap) addEvtHandler(obj interface{}) {
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		i.lggr.Error(
+			fmt.Errorf("informer expected configMap, got %v", obj),
+			"not forwarding event",
+		)
+		return
+	}
+	i.cmObjCh <- cm
+	i.bcaster.Action(watch.Added, cm)
+}
+
+func (i *InformerConfigMap) updateEvtHandler(oldObj, newObj interface{}) {
+	cm, ok := newObj.(*corev1.ConfigMap)
+	if !ok {
+		i.lggr.Error(
+			fmt.Errorf("informer expected configMap, got %v", newObj),
+			"not forwarding event",
+		)
+		return
+	}
+	i.cmObjCh <- cm
+	i.bcaster.Action(watch.Modified, cm)
+}
+
+func (i *InformerConfigMap) deleteEvtHandler(obj interface{}) {
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		i.lggr.Error(
+			fmt.Errorf("informer expected configMap, got %v", obj),
+			"not forwarding event",
+		)
+		return
+	}
+	i.cmObjCh <- cm
+	i.bcaster.Action(watch.Deleted, cm)
+}
+
+func (i *InformerConfigMap) GetCMObjectChannel() chan *corev1.ConfigMap {
```

Comment:
Would it be sufficient to return a read-only channel? i.e. `<-chan *corev1.ConfigMap` Also, can we rename this the full name, rather than `CM`?

```suggestion
func (i *InformerConfigMap) GetConfigMapObjectChannel() <-chan *corev1.ConfigMap {
```